scm-plugin/src/main/resources/archetype-resources/src/main/java/HelloWorldSCM.java

```diff
@@ -0,0 +1,50 @@
+package $package;
+
+import java.io.File;
+import java.io.IOException;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.scm.ChangeLogParser;
+import hudson.scm.SCM;
+import hudson.scm.SCMDescriptor;
+import hudson.scm.SCMRevisionState;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+
+public class HelloWorldSCM extends SCM {
+    @Override
+    public ChangeLogParser createChangeLogParser() {
+        return new HelloWorldSCMChangeLogParser();
+    }
+
+    @Override
+    public void checkout(@Nonnull Run<?, ?> build, @Nonnull Launcher launcher, @Nonnull FilePath workspace, @Nonnull TaskListener listener, File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
+    }
+
+
+    @Override
+    public SCMRevisionState calcRevisionsFromBuild(@Nonnull Run<?, ?> build, @Nullable FilePath workspace, @Nullable Launcher launcher, @Nonnull TaskListener listener) throws IOException, InterruptedException {
```

Comment:
You forgot `compareRemoteRevisionWith`, without which this is useless.

`createChangeLogParser` is also generally required. Generally, see [an example](https://github.com/jglick/jk--/tree/5223596aaa0c9bdc836d07969a73c15d23c34108/userspace-scm-plugin/src/main/java/org/jenkinsci/plugins/userspace_scm).

Reply:
`createChangeLogParser` exists. It's the second method.